src/llmq/quorums_dkgsessionhandler.cpp

```diff
@@ -95,18 +94,10 @@ CDKGSessionHandler::CDKGSessionHandler(const Consensus::LLMQParams& _params, ctp
     pendingJustifications((size_t)_params.size * 2, MSG_QUORUM_JUSTIFICATION),
     pendingPrematureCommitments((size_t)_params.size * 2, MSG_QUORUM_PREMATURE_COMMITMENT)
 {
-    phaseHandlerThread = std::thread([this] {
-        RenameThread(strprintf("dash-q-phase-%d", (uint8_t)params.type).c_str());
-        PhaseHandlerThread();
-    });
```

Comment:
I kind of liked the part which ensured that we use legit `LLMQType`. I think it still makes sense to check for `LLMQ_NONE` but maybe ctor is a better place for this i.e.
```
    if (params.type == Consensus::LLMQ_NONE) {
        throw std::runtime_error("Can't initialize CDKGSessionHandler with LLMQ_NONE type.");
    }
```

Thoughts?

Reply:
Im with you ;)

Reply:
IMHO there is no point in checking for invalid LLMQ type here. If it's invalid, there are hundreds of places where it would fail hard, e.g. due to std::map::at() uses.